xarray/core/computation.py

```diff
@@ -702,7 +713,8 @@ def stack(objects, dim, new_coord):
     elif any(is_dict_like(a) for a in args):
         return apply_dataset_ufunc(variables_ufunc, *args, signature=signature,
                                    join=join, exclude_dims=exclude_dims,
-                                   fill_value=dataset_fill_value)
+                                   fill_value=dataset_fill_value,
+                                   data_vars_join=data_vars_join)
```

Comment:
Add this argument to the `functools.partial` for `GroupBy` objects in the clause above this one.